### PR TITLE
Move eventid cut into uproot data loading

### PR DESCRIPTION
**Pull request by HenningSE**
This PR tries to move the changes from https://github.com/XENONnT/fuse/pull/154 into the data loading with uproot.

### CODE REVIEW

**Review by ramirezdiego — APPROVED** [review]

File: fuse/plugins/micro_physics/input.py [hunk 1]
```diff
@@ -389,29 +388,41 @@ def _load_root_file(self):
...
-        # If user specified entry start/stop we have to update number of
-        # events for source rate computation:
-        if self.kwargs["entry_start"] is not None:
-            start = self.kwargs["entry_start"]
+        # If we cut by eventid we have to read all of them first to find the start and stop index
+        if self.cut_by_eventid:
+            all_eventids = ttree.arrays("eventid")
+
+            if self.entry_start is not None:
+                start_index = np.argmax(all_eventids["eventid"] >= self.entry_start)
```
Comment (mhliu0001) [inline fuse/plugins/micro_physics/input.py:401]: I think you mean `start_index = np.argmax(all_eventids[all_eventids["eventid"] >= self.entry_start])`? Here you are performing `np.argmax` on a boolean array.

File: fuse/plugins/micro_physics/input.py [hunk 2]
```diff
@@ -389,29 +388,41 @@ def _load_root_file(self):
...
+            else:
+                start_index = 0
+
+            if self.entry_stop is not None:
+                stop_index = np.argmin(all_eventids["eventid"] < self.entry_stop)
```
Comment (mhliu0001) [inline fuse/plugins/micro_physics/input.py:406]: Same problem here.

Comment (mhliu0001) [reply fuse/plugins/micro_physics/input.py:406]: Also, here you need to make sure that all_eventids[all_eventids["eventid"] < self.entry_stop] is not empty. (Same for the former one)

File: fuse/plugins/micro_physics/input.py [hunk 3]
```diff
@@ -389,29 +388,41 @@ def _load_root_file(self):
...
+                if stop_index == 0:
```
Comment (mhliu0001) [inline fuse/plugins/micro_physics/input.py:407]: `stop_index == 0` can also happen when all events are in the range, in other words, `all_eventids["eventid"] < self.entry_stop` is all True.

Comment (mhliu0001) [reply fuse/plugins/micro_physics/input.py:407]: I think you mean `np.all(~(all_eventids["eventid"] < self.entry_stop))`?

**Pull request merged**